src/titiler/extensions/titiler/extensions/viewer.py

```diff
@@ -36,6 +36,8 @@ def cog_viewer(request: Request):
                     ),
                     "info_endpoint": factory.url_for(request, "info"),
                     "statistics_endpoint": factory.url_for(request, "statistics"),
+                    "viewer_enabled": hasattr(factory, "add_viewer")
+                    and getattr(factory, "add_viewer", False),
```

Comment:
I think that `getattr(class, "name", default)` should be enough 

> If the named attribute does not exist, default is returned if provided, otherwise [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) is raised. name need not be a Python identifier (see [setattr()](https://docs.python.org/3/library/functions.html#setattr)).